### PR TITLE
ON-1556 & ON-1684 & ON-1719 & ON-1758 : Gather apply information inject into assessor

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

**Comment by martin-hornsby** [comment]
I've added additional functionality to inject the organisation standard fi it is an initial application.

Also fixed a few issues injecting the appropriate Contact with the Organisation & Standard.

**This branch will likely need additional work to cater for ASLogin**

**Comment by martin-hornsby** [comment]
I've added functionality to inject the organisation standard when it is not an initial application.

**Comment by martin-hornsby** [comment]
**NOTE:** When we have EPAOrgId within Apply Organisation the following needs changing in `GatherAnswersForOrganisationStandardForApplication` 

```
var assessorOrganisation = (await _assessorApiClient.SearchOrganisations(organisation.Name)).FirstOrDefault();
var organisationId = assessorOrganisation?.Id;
```

**Comment by martin-hornsby** [comment]
ON-1758 added as it was apparent that FHA Details were not being injected into RoEPAO

File: src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs [hunk 1]
```diff
@@ -123,5 +123,15 @@ public void CheckIfUkprnAlreadyTaken(bool result)
...
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void IsOrganisationStandardTaken(bool result)
+        {
+            _registerValidationRepository.Setup(r => r.EpaOrganisationStandardExists(It.IsAny<string>(), It.IsAny<int>()))
+                .Returns(Task.FromResult(result));
+            var checkResult = _assessorValidationService.IsOrganisationStandardTaken("EPA0001", 1).Result;
+            Assert.AreEqual(checkResult, result);
+        }
```
Comment (davidgouge) [inline src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs]: How can both of those TestCases pass?

Comment (davidgouge) [reply src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs]: Oh I see, the .Setup has it's return set to the expected.  Not sure this is a very good test, but not sure why.

Comment (davidgouge) [reply src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs]: Would the better test be to make sure that the correct values get passed through to EpaOrganisationStandardExists? 
And check the return from EpaOrganisationStandardExists gets returned from IsOrganisationStandardTaken is the same (which i guess is what this test is doing, but it doesn't need to do both test cases).

File: src/SFA.DAS.AssessorService.Data/RegisterQueryRepository.cs [hunk 2]
```diff
@@ -374,6 +373,40 @@ public async Task<EpaContact> GetContactByContactId(Guid contactId)
...
+
+                var sql =
+                    "select Id, EndPointAssessorOrganisationId, Username,GivenNames, DisplayName, FamilyName, SigninId, SigninType, Email, Status, PhoneNumber " +
+                    " from Contacts where Email = @email";
+                var contact = await connection.QuerySingleOrDefaultAsync<EpaContact>(sql, new { email });
+                return contact;
+            }
+        }
+
+        public async Task<EpaContact> GetContactBySignInId(string signinId)
```
Comment (davidgouge) [inline src/SFA.DAS.AssessorService.Data/RegisterQueryRepository.cs]: Why is this returning an EpaContact and not a Contact?  And surely we have a GetContactBySignInId somewhere else?  Same for Email as well I guess?

File: src/SFA.DAS.AssessorService.Web.Staff/Controllers/Apply/ApplicationController.cs [hunk 3]
```diff
@@ -258,12 +262,29 @@ public async Task<IActionResult> Return(Guid applicationId, int sequenceId, stri
             var warningMessages = new List<string>();
             if (sequenceId == 2 && returnType == "Approve")
             {
-                var command = await _answerService.GatherAnswersForOrganisationAndContactForApplication(applicationId);
-                var response = await _answerInjectionService.InjectApplyOrganisationAndContactDetailsIntoRegister(command);
-                warningMessages = response.WarningMessages;
-                if (warningMessages.Count == 0 && !response.IsEpaoApproved)
+                var sequneceOne = await _applyApiClient.GetSequence(applicationId, 1);
```
Comment (davidgouge) [inline src/SFA.DAS.AssessorService.Web.Staff/Controllers/Apply/ApplicationController.cs]: Typo. :)

Comment (martin-hornsby) [reply src/SFA.DAS.AssessorService.Web.Staff/Controllers/Apply/ApplicationController.cs]: Good spot!

File: src/SFA.DAS.AssessorService.Data/RegisterQueryRepository.cs (hunk 2 above)
Comment (martin-hornsby) [reply src/SFA.DAS.AssessorService.Data/RegisterQueryRepository.cs]: Probably as Mark said, that `EPA distinguished itself as being dappery, when there was a lot of EF still in use`
Assuming it's going to be a bit of a job to clean this up into 1 common object again

File: src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs (hunk 1 above)
Comment (martin-hornsby) [reply src/SFA.DAS.AssessorService.Application.Api.UnitTests/Services/AssessorValidationServiceTests.cs]: It follows the same pattern as the other tests in here (i.e. true / false)

I read these tests as the registerValidationRepository will return the appropraite response and thus assessorValidationService should return that response too (i.e. it bubbles up through the relevant layers in the application).

**Pull request merged**